From 1da3299aecc56a1ba590a7fb7ae868275bb46d93 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?J=C3=B6rg=20Behrmann?= <behrmann@physik.fu-berlin.de>
Date: Thu, 27 Aug 2026 16:53:27 +0200
Subject: [PATCH] tests: skip test_fork_and_wait if on rc 1

This is relevant for restricted build environments where unshare could get an
EPERM. Unfortunately we only get the rc from the child in fork_and_wait back
and this could be any return code of 1, but anything but EPERM for unshare is
unlikely here.
---
 tests/test_run.py | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/tests/test_run.py b/tests/test_run.py
index 52b43067..0d588a3a 100644
--- a/tests/test_run.py
+++ b/tests/test_run.py
@@ -8,6 +8,7 @@ from pathlib import Path
 import pytest
 
 from mkosi.run import fork_and_wait
+from mkosi.sandbox import EPERM
 
 
 def test_fork_and_wait_returns_value() -> None:
@@ -60,5 +61,10 @@ def test_fork_and_wait_sandbox(tmp_path: Path) -> None:
     def exists() -> bool:
         return Path("/abc").exists()
 
-    result = fork_and_wait(exists, sandbox=contextlib.nullcontext(["--bind", os.fspath(tmp_path), "/"]))
+    try:
+        result = fork_and_wait(exists, sandbox=contextlib.nullcontext(["--bind", os.fspath(tmp_path), "/"]))
+    except subprocess.CalledProcessError as e:
+        if e.returncode == EPERM or e.returncode == 28:
+            pytest.skip("CLONE_NEWUSER is not allowed in the test environment")
+        raise
     assert result
-- 
2.51.0

